### DISCUSSION TITLE
Are leases linearizable?

### DISCUSSION THREAD

**Discussion by mladedav** [etcd-io/etcd#15543] [Q&A] [answered]
The docs mention that watch is not linearizable and range can optionaly not be linearizable. I got under the impression that everything else is linearizable.

Is my understanding wrong and is in fact just put & range linearizable and the rest is only sequencable?

My question stems from some failed tests on our etcd client where we create 3 leases in succession (always waiting for the response and checking that the ID and TTL match what we expect) and then we request all leases and the last was missing.

I have also noticed that granting and revoking leases does not change revision numbers which I guess is ok because order matters only when a lease is tied to a key-value which changes revision. However, does that also imply that it is possibly not even sequencable if we can create one lease at one node, another lease at another and then if someone calls LeaseLeases then they may get either the first, the second, or both leases, but at this point there could even be a scenario where one node returns just the first lease and another node just the second one?

**Comment by serathius** [comment, marked as answer]
Nope, or at least not as of v3.5 release https://github.com/etcd-io/etcd/issues/13915
